src/core/NotePlayHandle.cpp

```diff
@@ -160,6 +160,8 @@ void NotePlayHandle::done()
 	if( buffer() ) releaseBuffer();
 
 	unlock();
+
+	this->PlayHandle::~PlayHandle();
```

Comment:
It should never be necessary to call destructors explicitly in C++. If `PlayHandle`'s destructor is never called to release the buffer, something else is wrong here - if it is called, this patch will lead to the buffer being freed twice.

Reply:
@lukas-w Usually yes, but I think *no* in this situation. Since `NotePlayHandle` uses memory allocated by `NotePlayHandleManager` and its destructor is never called, `NotePlayHandle::done()` should do what destructor does.
If you really don't like this, you may consider redesigning or removing `NotePlayHandleManager`.

Reply:
Note: https://stackoverflow.com/questions/14187006/is-calling-destructor-manually-always-a-sign-of-bad-design

Reply:
Then it's a bug in `NotePlayHandleManager`. The destructor should be called in `NotePlayHandleManager::release` because it's `NotePlayHandleManager::acquire` where placement-new is used. Also, it looks like `NotePlayHandle::done()`'s code actually belongs in `~NotePlayHandle`.

Reply:
Then renaming `NotePlayHandle::done()` to `NotePlayHandle::~NotePlayHandle()` and calling it in the `NotePlayHandleManager::release` would just be fine?

Reply:
Yep, I think it's fine this way 👍 As far as I can see, `done()` is only called in `NotePlayHandleManager::release()` and simply plays the role of a destructor.